tensorflow/compiler/xla/types.h

```diff
@@ -45,6 +45,12 @@ using complex128 = std::complex<double>;
 
 using ::Eigen::half;
 
+namespace gpu {
```

Comment:
xla/types.h is for backend-independent code only, so namespace gpu shouldn't appear here. With my suggestion on GpuExecutable constructor above, I don't think this is needed in this header anymore.

Reply:
would it be feasible to put `GpuVersion` under `xla` namespace then? Looking at the skeleton code in `mlir_gpu` I could imagine this type be used later on in that module.

Reply:
It's not about the namespace, it's about the directory. `compiler/xla/service/gpu` is the place for xla::gpu::GpuVersion. Maybe just define it in `gpu_executable.h`?

Reply:
@timshen91  `gpu_executable.h` may not be a 100% good place, because this type is also used by `//tensorflow/compiler/xla/service/gpu/llvm_gpu_backend` target, which doesn't really depend on `//tensorflow/compiler/xla/service/gpu:gpu_executable`:

https://github.com/tensorflow/tensorflow/blob/master/tensorflow/compiler/xla/service/gpu/llvm_gpu_backend/BUILD#L16

In fact, looking at common header files used by `nvptx_backend_lib.cc` and `nvptx_compiler.cc` it turns out `//tensorflow/compiler/xla:types` be a common dependent target.

That said, I'm open to moving this type alias to `gpu_executable.h`, but would you feel comfortable if we add an extra dependency of `//tensorflow/compiler/xla/service/gpu:gpu_executable` in `//tensorflow/compiler/xla/service/gpu/llvm_gpu_backend`?

Reply:
I mean, in the worst case we can create a new file, but GPU-related code has to live in `service/gpu`. I think we can create a new file `service/gpu/gpu_types.h`.

Reply:
Sounds good to me, will create service/gpu/gpu_types.h